atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractGetCandidateCellsForSweepingTest.java

```diff
@@ -0,0 +1,242 @@
+    }
+
+    @Test
+    public void doNotReturnCandidateIfOnlyCommittedTimestamp() {
+        new TestDataBuilder().put(1, 1, 10L).store();
+        assertThat(getAllCandidates(conservativeRequest(PtBytes.EMPTY_BYTE_ARRAY, 40L, 30L))).isEmpty();
+    }
+
+    @Test
+    public void returnCandidateIfTwoCommittedTimestamps() {
+        new TestDataBuilder().put(1, 1, 10L).put(1, 1, 20L).store();
+        assertThat(getAllCandidates(conservativeRequest(PtBytes.EMPTY_BYTE_ARRAY, 40L, 30L)))
+                .containsExactly(ImmutableCandidateCellForSweeping.builder()
+                        .cell(cell(1, 1))
+                        .sortedTimestamps(new long[] { 10L, 20L })
+                        .isLatestValueEmpty(false)
+                        .numCellsTsPairsExamined(2)
+                        .build());
+    }
+
+    @Test
+    public void doNotReturnCandidateWithCommitedEmptyValueIfConservative() {
+        new TestDataBuilder().putEmpty(1, 1, 10L).store();
+        assertThat(getAllCandidates(conservativeRequest(PtBytes.EMPTY_BYTE_ARRAY, 40L, 30L))).isEmpty();
+    }
+
+    @Test
+    public void returnCandidateWithCommitedEmptyValueIfThorough() {
+        new TestDataBuilder().putEmpty(1, 1, 10L).store();
+        assertThat(getAllCandidates(thoroughRequest(PtBytes.EMPTY_BYTE_ARRAY, 40L, 30L)))
+                .containsExactly(ImmutableCandidateCellForSweeping.builder()
+                        .cell(cell(1, 1))
+                        .sortedTimestamps(new long[] { 10L })
+                        .isLatestValueEmpty(true)
+                        .numCellsTsPairsExamined(1)
+                        .build());
+    }
+
+    @Test
+    public void returnCellsInOrder() {
+        new TestDataBuilder()
+                .putEmpty(1, 1, 10L)
+                .putEmpty(1, 2, 10L)
+                .putEmpty(2, 2, 10L)
+                .putEmpty(3, 1, 10L)
+                .putEmpty(3, 2, 10L)
+                .putEmpty(3, 3, 10L)
+                .store();
+        assertThat(getAllCandidates(conservativeRequest(PtBytes.EMPTY_BYTE_ARRAY, 30L, 5L))
+                    .stream().map(CandidateCellForSweeping::cell).collect(Collectors.toList()))
+                .containsExactly(cell(1, 1), cell(1, 2), cell(2, 2), cell(3, 1), cell(3, 2), cell(3, 3));
+    }
+
+    @Test
+    public void startFromGivenRow() {
+        new TestDataBuilder()
+                .putEmpty(1, 1, 10L)
+                .putEmpty(1, 2, 10L)
+                .putEmpty(2, 1, 10L)
+                .putEmpty(2, 2, 10L)
+                .putEmpty(3, 1, 10L)
+                .putEmpty(3, 2, 10L)
+                .store();
+        assertThat(getAllCandidates(conservativeRequest(cell(2, 2).getRowName(), 30L, 5L))
+                .stream().map(CandidateCellForSweeping::cell).collect(Collectors.toList()))
+                .containsExactly(cell(2, 1), cell(2, 2), cell(3, 1), cell(3, 2));
+    }
+
+    @Test
+    public void largerTableWithSmallBatchSizeReturnsCorrectResults() {
```

Comment:
The problem is that the batch hint is only a hint and the exact batch size is implementation-defined, so it's unclear how to write that kind of test correctly (i.e. without coding against the implementation).